File: project/ProbusClient/app/src/main/java/dev/malikkurosaki/probusclient/MainActivity.java
package dev.malikkurosaki.probusclient;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.widget.TextView;

import java.util.List;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class MainActivity extends AppCompatActivity {

    private TextView result;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);


        result = findViewById(R.id.result);

        Retrofit retrofit = new Retrofit.Builder()
                .baseUrl("https://103.207.96.250:443/")
                .addConverterFactory(GsonConverterFactory.create())
                .build();

        JsonPlaceHolderApi jsonPlaceHolderApi = retrofit.create(JsonPlaceHolderApi.class);
        Call<List<Post>> call = jsonPlaceHolderApi.getPost();
        call.enqueue(new Callback<List<Post>>() {
            @Override
            public void onResponse(Call<List<Post>> call, Response<List<Post>> response) {
                if (!response.isSuccessful()){
                    result.setText(response.code());
                    return;
                }

                /*List<Post> posts = response.body();
                for (Post post : posts ){
                    String csontent = "";
                    csontent += post.getText() +"\n\n\n";

                    result.append(csontent);
                }*/
            }

            @Override
            public void onFailure(Call<List<Post>> call, Throwable t) {
                result.setText(t.getMessage());
            }
        });
    }
}

File: project2/KotlinV1/app/src/main/java/dev/malikkurosaki/kotlinv1/MainActivity.kt
package dev.malikkurosaki.kotlinv1

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.util.Log
import kotlinx.android.synthetic.main.activity_main.*
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory
import retrofit2.http.GET

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        /*val userInterface = DataRepository.create()
        userInterface.getPost().enqueue(object : Callback<List<User>> {
            override fun onFailure(call: Call<List<User>>, t: Throwable) {
                Log.d("aaaa",t.toString())
            }

            override fun onResponse(call: Call<List<User>>, response: Response<List<User>>) {

                val data = response.body();

                data?.map {
                    isian.append("${it.name} \n")
                }
            }

        })*/


        val inter = Kurungan.create()
        inter.getNama().enqueue(object :Callback<List<Nama>>{
            override fun onFailure(call: Call<List<Nama>>, t: Throwable) {
                Log.d("gangguan",t.toString())
            }

            override fun onResponse(call: Call<List<Nama>>, response: Response<List<Nama>>) {
                val data = response.body()

                data?.map {
                    isian.append("${it.name} \n")
                }
            }

        })
    }

    data class Nama(val id:String,val name:String);

    interface NamaUser{
        @GET("user")
        fun getNama():Call<List<Nama>>
    }

    object Kurungan{
        fun create():NamaUser{
            val ret = Retrofit.Builder()
                .addConverterFactory(GsonConverterFactory.create())
                .baseUrl("https://radiant-caverns-42043.herokuapp.com/")
                .build()
            return ret.create(NamaUser::class.java)
        }
    }
}
